solr/solrj/src/java/org/apache/solr/client/solrj/impl/Http2SolrClient.java

```diff
@@ -1273,4 +1287,33 @@ static SslContextFactory.Client getDefaultSslContextFactory() {
 
     return sslContextFactory;
   }
+
+  private static class MDCCopyHelper extends RequestResponseListener {
```

Comment:
it felt like a good representation of the lifecycle I needed to hook into, so I used it. in the end it turned out it's not stritly needed but I liked the idea of reusing an existing model instead of inventing a new set of names.

Reply:
Okay then a comment to this affect would be helpful.